### PR TITLE
Add generic annotation hook

### PR DESCRIPTION
**Pull request by ryanjduffy**
Exposes `__RECORD_REPLAY_ANNOTATION_HOOK__` to allow userland scripts to add annotations to replays

### CODE REVIEW

**Comment by kannanvijayan** [comment]
This looks good actually, have you tested it?

**Comment by ryanjduffy** [comment]
No. I wanted to check in on it and then see about the build process since I haven't tried it before (and I don't have a linux box locally on which to build).

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 1]
```diff
@@ -25,6 +25,8 @@ static const char DirectorySeparator = '/';
 static const char DirectorySeparator = '\\';
 #endif
 
+#define ANNOTATION_HOOK_NAME "__RECORD_REPLAY_ANNOTATION_HOOK__"
```
Comment (bhackett1024) [inline third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:28]: This should be a `static const char*` to minimize `#define` usage.

**Review by bhackett1024 — APPROVED** [review]

**Comment by kannanvijayan** [comment]
I landed the v8 changes.  Is there any reason to wait for this?  I think we can land and do a browser release.

**Pull request merged**